src/ui/SearchInterface/SearchInterface.ts

```diff
@@ -763,6 +778,49 @@ export class SearchInterface extends RootComponent implements IComponentBindings
     return QueryStateModel.defaultAttributes.quickview;
   }
 
+  private handleFacetValueState(stateToSet: { [key: string]: any }, fvFields: string[]): void {
+    const facetRef = BaseComponent.getComponentRef('Facet');
```

Comment:
> Why not interact with the actual facet component instead

Sure

> It can be processed like any other events.

Isn't the purpose of `preprocess` to alter the state before setting it? This state type is used to have one of two behavior depending on the context and is then removed.

> I think it would be much cleaner to have a dedicated class for this

Sure :)